app/services/ssm_parameters.rb

```diff
@@ -18,6 +18,15 @@ def delete_parameter
                              })
   end
 
+  def get_invalid_parameters(ssm_paths)
+    return [] if ssm_paths.empty?
+
+    response = client.get_parameters({
+                                       names: ssm_paths
+                                     })
+    response.invalid_parameters
+  end
```

Comment:
The problem with using outside APIs in such a raw way is that they can throw all sorts of errors you do not care about or throw errors because of parameters that you did not expect they could not handle. I suggest putting a catch-all rescue here and logging the exception.

```suggestion
    response.invalid_parameters
  rescue StandardError => e
    Rails.logger.error("Unexpected error #{e}")
  end
```